src/contest/weglide/UploadIGCFile.cpp

```diff
@@ -0,0 +1,207 @@
+  BrokenDate birthdate;
+  StaticString<0x80> name;
+};
+
+struct Aircraft {
+  uint32_t id = 0;
+  StaticString<0x40> name;
+  StaticString<4> kind;  // 'MG' - motor aircraft,...
+  StaticString<10> sc_class;
+};
+
+struct UploadResponse {
+  uint64_t flight_id = 0;
+  User user;
+  Aircraft aircraft;
+  StaticString<0x40> scoring_date;
+  StaticString<0x40> registration;
+  StaticString<0x40> competition_id;
+};
+
+static void
+UploadJsonInterpreter(const boost::json::value &json,
+                      UploadResponse& response) {
+  if (json.is_array() && (json.get_array().size() > 0)) {
+    auto flight_obj = json.get_array().at(0);
+    response.scoring_date = GetJsonString(flight_obj, "scoring_date").c_str();
+    response.flight_id = flight_obj.at("id").to_number<int64_t>();
+    response.registration = GetJsonString(flight_obj, "registration").c_str();
+    response.competition_id = GetJsonString(flight_obj, "competition_id").c_str();
+    auto user = flight_obj.at("user");
+    if (user.is_object()) {
+      auto user_obj = user.as_object();
+      response.user.id = user_obj.at("id").to_number<uint32_t>();
+      response.user.name = GetJsonString(user_obj, "name").c_str();
+    }
+    auto aircraft = flight_obj.at("aircraft");
+    if (aircraft.is_object()) {
+      auto aircraft_obj = aircraft.as_object();
+      response.aircraft.id = aircraft_obj.at("id").to_number<uint32_t>();
+      response.aircraft.name = GetJsonString(aircraft_obj, "name").c_str();
+      response.aircraft.kind = GetJsonString(aircraft_obj, "kind").c_str();
+      response.aircraft.sc_class = GetJsonString(aircraft_obj, "sc_class").c_str();
+    }
+  }
+  else {
+    throw std::runtime_error("The http response has a wrong JSON format!");
+  }
+}
+
+static void
+UploadSuccessDialog(const UploadResponse& response, const Path igc_path) {
+  // TODO: Create a real Dialog with fields
+  // With this Dialog insert the possibilty to update/patch the flight
+  // f.e. copilot in double seater, scoring class, short comment and so on
+  StaticString<0x1000> msg;
+  msg.Format(_("File upload '%s' was succesfull"),
+    igc_path.c_str());
+  msg.append(_T("!\n\n"));
+  msg.AppendFormat(_T("%-20s: %u\n"), _("flight_id"), response.flight_id);
+  msg.AppendFormat(_T("%-20s: %s\n"), _("scoring_date"),
+    response.scoring_date.c_str());
+  msg.AppendFormat(_T("%-20s: %s (%d)\n"), _("user"),
+    response.user.name.c_str(), response.user.id);
+  msg.AppendFormat(_T("%-20s: %s (%u)\n"), _("aircraft type"),
+    response.aircraft.name.c_str(), response.aircraft.id);
+  msg.AppendFormat(_T("%-20s: %s: %s "), _("aircraft"), _("reg"),
+    response.registration.c_str());
+  msg.AppendFormat(_T("%s: %s\n"), _("cid"),
+    response.competition_id.c_str());
+
+  ShowMessageBox(msg, _T("WeGlide Upload"), MB_OK | MB_ICONEXCLAMATION);
+}
+
+static Co::InvokeTask
+UpdateTask(Path igc_path, const WeGlideSettings& settings,
+  uint_least32_t glider_id, boost::json::value& json,
```

Comment:
No output parameters! If you want to return a value, return a value.

Reply:
Ok, I have to check it with this CoRoutines...
This isn't really clear to me, but I want to verify.

Reply:
Done with a simple structure, but I don't know if it's enough!
The good thing: It works!